strategies/affected/src/main/java/org/arquillian/smart/testing/strategies/affected/ClassDependenciesGraph.java

```diff
@@ -77,14 +83,86 @@ void buildTestDependencyGraph(Collection<File> testJavaFiles) {
 
         // Then find dependencies
         for (String changedTestClassNames : testClassesNames) {
-            JavaClass javaClass = builder.getClassDescription(changedTestClassNames);
-            if (javaClass != null) {
-                addToIndex(new JavaElement(javaClass), javaClass.getImports());
+            JavaClass testJavaClass = builder.getClassDescription(changedTestClassNames);
+            if (testJavaClass != null) {
+                final String[] imports = testJavaClass.getImports();
+                final List<String> manualProductionClasses = calculateManualAddedDependencies(testJavaClass);
+                manualProductionClasses.addAll(Arrays.asList(imports));
+                addToIndex(new JavaElement(testJavaClass), manualProductionClasses);
             }
         }
     }
 
-    private void addToIndex(JavaElement javaElement, String[] imports) {
+    private List<String> calculateManualAddedDependencies(JavaClass testJavaClass) {
+        final List<String> manualDependencyClasses = new ArrayList<>();
+        final ComponentUnderTest[] allTestsAnnotation = getAllAnnotations(testJavaClass);
+
+        for (ComponentUnderTest tests : allTestsAnnotation) {
+            List<String> packages = getPackages(testJavaClass.packageName(), tests);
+            for (String pkg : packages) {
+                final String trimmedPackage = pkg.trim();
+                manualDependencyClasses.addAll(scanClassesFromPackage(trimmedPackage));
+            }
+        }
+
+        return manualDependencyClasses;
+
+    }
+
+    private ComponentUnderTest[] getAllAnnotations(JavaClass testJavaClass) {
```

Comment:
Minor detail - fix it when you will find time - instead of generic `getAllAnnotations` - `findComponentsUnderTests`?